File: leetcode/1598-crawler-log-folder/minOperations.test.ts
import cases from "./cases.json";
import minOperations from "./minOperations";

test("crawler log folder", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    expect(minOperations(testCase.input)).toEqual(testCase.output);
  });
});

File: leetcode/1598-crawler-log-folder/minOperations_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type MinOperationsCase struct {
	Input []string

	Output int
}

func TestMinOperations(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MinOperationsCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("nums: %v ==> %v \n", testCase.Input, testCase.Output)

		result := minOperations(testCase.Input)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
